### ISSUE TITLE
BoxDownload#getOutputFile returns null

### ISSUE THREAD

**Issue by MariusVolkhart** [box/box-android-content-sdk#14]
`BoxDownload#getOutputFile()` is documented as returning
> the output file for the download.

But it has a return value of `null` hard coded.

Consider the following snippet. The assertion at the bottom will fail, though it seems semantically reasonable and the documentation does not suggest it to be incorrect.
```java
BoxApiFile boxApiFile;
File downloadTarget;
String fileId;

// Some setup...

File sameAsDownloadTarget = boxApiFile.getDownloadRequest(downloadTarget, fileId).send().getOutputFile();

assert downloadTarget.equals(sameAsDownloadTarget);
```

https://github.com/box/box-android-content-sdk/blob/master/box-content-sdk/src/main/java/com/box/androidsdk/content/models/BoxDownload.java#L94

**Comment by doncung** [comment]
That's odd, the method in BoxDownload is overwritten in BoxRequestDownload to return the file provided or a newly created file inside the directory provided.

**Comment by doncung** [comment]
I tried a test similar to yours and getOutputFile does return a File for me. If you're getting something different can you post  your full source? Might be a particular file.

**Comment by MariusVolkhart** [comment]
I gave you the wrong example code. We were using the variant of `BoxApiFile#getDownloadRequest()` that takes a `OutputStream`. I figured I should still be able to get the output file but there's not really way to do so.

I would prefer to see an `IllegalStateException` instead of returning `null` if calling `getOutputFile()` on an object of the stream variant, but this issue is definitely closed.
